### DISCUSSION TITLE
Change from `IfLetStore` to `if let` produces wrong type of store in `observation-beta` refactor

### DISCUSSION THREAD

**Discussion by appfrosch** [pointfreeco/swift-composable-architecture#2657] [Beta]
I am trying to refactor a working 3 panel `NavigationSplitView` to use the new `observation-beta` enhancements.
I was able to change everything except the `IfLetStore`–when I try to refactor that to use the vanilla `if let`, it produces the wrong type of store.

Here's the diff:

```swift
struct AppView: View {
   var body: some View {
     Group {
       NavigationSplitView {
-        IfLetStore(
-          self.store.scope(
-            state: \.$sidepanel,
-            action: \.sidepanel
-          )
-        ) { store in
-          SidepanelView(store: store)
-            .padding()
+        if let sidepanelStore = store.scope(state: \.sidepanel, action: \.sidepanel) {
+          SidepanelView(store: sidepanelStore)
         }
       } content: {
         Group  {
//…
```

The error message is as follows: `Cannot convert value of type 'Store<SidepanelFeature.State, PresentationAction<SidepanelFeature.Action>>' to expected argument type 'StoreOf<SidepanelFeature>' (aka 'Store<SidepanelFeature.State, SidepanelFeature.Action>')`.

The `sidepanel` is declared like so: `@Presents var sidepanel: SidepanelFeature.State?`, the `SidepanelView` looks like this:

```swift
struct SidepanelView: View {
  let store: StoreOf<SidepanelFeature>

  var body: some View {
    Form {
      Button("Load Sample Data") {
        store.send(.delegate(.showSampleDataButtonPressed))
      }
      Spacer()
    }
    .frame(minWidth: 150)
    .padding()
  }
}
```

What am I missing/doing wrong here?

I created a working sample [in this repo](https://github.com/appfrosch/so-tcaThreePanelObservationBeta)–any hints are highly appreciated 😀

**Comment by mbrandonw** [comment]
Hi @appfrosch, I didn't pull down your code, but does it work if you change the scope to this:

```swift
if let sidepanelStore = store.scope(state: \.sidepanel, action: \.sidepanel.presented) {
  …
}
```

This is necessary because your `sidepanel` holds onto `PresentationAction`, and so you have to further dive into the `presented` case of that.

We should add some documentation on this.

**Reply by appfrosch** [reply]
Yep, it sure does-thanks a lot.
Amazes me every single time, how responsive and helpful you guys are. 

Thanks for an amazing framework (that I still have to grasp the concepts of even better), your wonderful support and your video series (even though more often I'd like to admit it goes over my head...).

Happy holiday season to the both of you!
